spec/Scoringline/SendinblueApi/Api/EmailSpec.php

```diff
@@ -0,0 +1,167 @@
+        TransformerInterface $transformer
+    ) {
+        $result = [
+            'code' => 'success',
+            'message' => 'Email sent successfully',
+            'data' => []
+        ];
+
+        $resultString = json_encode($result);
+        $client->send(Argument::any())->willReturn($resultString);
+        $transformer->transform($resultString)->willReturn($result);
+
+        $this->sendSimpleEmail(
+            ['from@example.com', 'from name!'],
+            ['to@example.com' => 'to name!'],
+            'Invitation',
+            '<h1>HTML</h1> content'
+        )->shouldReturn($result);
+    }
+
+    function it_should_send_advance_email_with_attachment_and_inline_image_when_exists_files(
+        AbstractHttpClient $client,
+        TransformerInterface $transformer,
+        Email $email
+    ) {
+        $result = [
+            'code' => 'success',
+            'message' => 'Email sent successfully',
+            'data' => []
+        ];
+
+        $file = new File('fixtures/test.txt');
+        $imageFile = new File('fixtures/logo_one.png');
+
+        $resultString = json_encode($result);
+        $client->send(Argument::any())->willReturn($resultString);
+        $transformer->transform($resultString)->willReturn($result);
+
+        $email->setTo(['to@example.com' => 'to name!']);
+        $email->setFrom(['from@example.com', 'from name!']);
+        $email->setSubject('Invitation');
+        $email->setText('You are invited for giving test');
+        $email->setHtml('This is the <h1>HTML</h1>');
+        $email->setAttachments([$file, 'fixtures/logo.png']);
+        $email->setInlineImages(['fixtures/logo.png', $imageFile]);
+
+        $this->sendEmail($email)->shouldReturn($result);
+    }
+
+    function it_should_throw_exception_when_email_send_failed_due_to_missing_params(
+        AbstractHttpClient $client,
+        TransformerInterface $transformer,
+        Email $email
+    ) {
+        $result = [
+            'code' => 'failure',
+            'message' => 'Unable to send email. Exception message was: No recipient forward path has been supplied.',
+            'data' => []
+        ];
+
+        $resultString = json_encode($result);
+        $emailFailureException = new EmailSendFailureException($resultString);
+        $client->send(Argument::any())->willThrow($emailFailureException);
+        $transformer->transform($resultString)->willThrow($emailFailureException);
+
+        $email->setFrom(['from@example.com', 'from name!']);
+        $email->setSubject('Hello');
+        $email->setHtml('Hi this is test <h1>email</h1>');
+
+        $this
+            ->shouldThrow($emailFailureException)
+            ->duringSendEmail($email)
+        ;
+
+        $this->shouldThrow($emailFailureException)
+            ->duringSendSimpleEmail(['joni@sendinblue.com', 'Joni'], [], 'Invite', 'Content')
+        ;
+
+        $this->shouldThrow('PhpSpec\Exception\Example\ErrorException')
```

Comment:
You're doing it just like volkswagen ! http://www.bbc.com/news/business-34324772 :D

Your code look like this plugin: https://github.com/hmlb/phpunit-vw